### PR TITLE
https://servicecomb.atlassian.net/browse/JAV-501

### PR DESCRIPTION
**Pull request by coolhongluo**
As a developer, I need to have authentication in service provider to protect the microservice from unauthorized access

### CODE REVIEW

File: foundations/foundation-common/src/main/java/io/servicecomb/foundation/common/utils/RSAKeyPairEntry.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package io.servicecomb.foundation.common.utils;
```
Comment (liubao68) [inline foundations/foundation-common/src/main/java/io/servicecomb/foundation/common/utils/RSAKeyPairEntry.java]: code template is not correct, please check etc folder for eclipse and IDEA code template

File: foundations/foundation-common/src/main/java/io/servicecomb/foundation/common/utils/RSAUtils.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  private final static String SIGN_ALG = "SHA256withRSA";
+
+  private static Base64.Encoder encoder = Base64.getEncoder();
+
+  private static Base64.Decoder decoder = Base64.getDecoder();
+
+  public static RSAKeyPairEntry getRSAKeyPair() {
+    try {
+      KeyPairGenerator keyGenerator = KeyPairGenerator.getInstance(RSA_ALG);
+      keyGenerator.initialize(1024, new SecureRandom());
```
Comment (liubao68) [inline foundations/foundation-common/src/main/java/io/servicecomb/foundation/common/utils/RSAUtils.java]: key size is too short

File: foundations/foundation-common/src/test/java/io/servicecomb/foundation/common/utils/TestRollingFileAppenderExt.java [hunk 3]
```diff
@@ -22,6 +22,7 @@
 import org.apache.log4j.spi.LoggingEvent;
 import org.junit.Assert;
 import org.junit.Test;
+import org.junit.runners.Parameterized.Parameters;
```
Comment (liubao68) [inline foundations/foundation-common/src/test/java/io/servicecomb/foundation/common/utils/TestRollingFileAppenderExt.java]: unused imports, please clear all compile warnings.

File: handlers/handler-publickey-auth/src/main/java/io/servicecomb/authentication/consumer/ConsumerAuthHandler.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  private RSACoumserTokenManager athenticationTokenManager = new RSACoumserTokenManager();
+
+  @Override
+  public void handle(Invocation invocation, AsyncResponse asyncResp) throws Exception {
+
+    Optional<String> token = Optional.ofNullable(athenticationTokenManager.getToken());
+    if(!token.isPresent())
+    {
+      asyncResp.consumerFail(
+          new Error("rejected by consumer authentication handler"));
```
Comment (liubao68) [inline handlers/handler-publickey-auth/src/main/java/io/servicecomb/authentication/consumer/ConsumerAuthHandler.java]: Do not throw Error for application invoke

File: foundations/foundation-common/src/test/java/io/servicecomb/foundation/common/utils/TestRollingFileAppenderExt.java (hunk 3 above)
Comment (coolhongluo) [reply foundations/foundation-common/src/test/java/io/servicecomb/foundation/common/utils/TestRollingFileAppenderExt.java]: accept

File: foundations/foundation-common/src/main/java/io/servicecomb/foundation/common/utils/RSAUtils.java (hunk 2 above)
Comment (coolhongluo) [reply foundations/foundation-common/src/main/java/io/servicecomb/foundation/common/utils/RSAUtils.java]: accept ，change to 2048

File: handlers/handler-publickey-auth/src/main/java/io/servicecomb/authentication/consumer/ConsumerAuthHandler.java (hunk 4 above)
Comment (coolhongluo) [reply handlers/handler-publickey-auth/src/main/java/io/servicecomb/authentication/consumer/ConsumerAuthHandler.java]: throw new Exception?

File: foundations/foundation-common/src/main/java/io/servicecomb/foundation/common/utils/RSAKeyPairEntry.java (hunk 1 above)
Comment (coolhongluo) [reply foundations/foundation-common/src/main/java/io/servicecomb/foundation/common/utils/RSAKeyPairEntry.java]: accept , forget to format

**Pull request closed without merging**